tests/vm_test.go

```diff
@@ -528,11 +528,16 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			vm := createRunningVM(virtClient, vmi)
 
 			By("Verifying that the status toggles between ErrImagePull and ImagePullBackOff")
-			const times = 2
-			for i := 0; i < times; i++ {
-				Eventually(ThisVM(vm), 300*time.Second, 1*time.Second).Should(HavePrintableStatus(v1.VirtualMachineStatusErrImagePull))
-				Eventually(ThisVM(vm), 300*time.Second, 1*time.Second).Should(HavePrintableStatus(v1.VirtualMachineStatusImagePullBackOff))
+			maxImgPullRetries := 2 // max number of image pull (re)tries
+			expectedStates := []v1.VirtualMachinePrintableStatus{}
+			for range maxImgPullRetries {
+				// each image pull (re)try should take the VM
+				// from status ErrImagePull to ImagePullBackOff
+				expectedStates = append(expectedStates, v1.VirtualMachineStatusErrImagePull)
+				expectedStates = append(expectedStates, v1.VirtualMachineStatusImagePullBackOff)
 			}
+
+			waitForVMStateTransition(virtClient, vm, expectedStates, 600*time.Second)
```

Comment:
Can we lower the timeout?

Reply:
Previously this test had 4 timeout checks, each 5 mins long. 5 mins is the longest backoff duration in K8s. 
In its updated form, the testcase code does not guarantee exactly when in the ` `... -> ErrImagePull -> ImagePullBackOff -> ...` state-transition sequence the Watch starts. So I am assuming the worst case scenario, where the max backoff grace period (5 mins) is used.
Since we need to verify one full transition from `ErrImagePull -> ImagePullBackOff`, we should wait atleast 5 mins plus the time taken to attempt image pull twice and other K8s/KubeVirt overheads. I can change the timeout to be 6 minutes to account for that.

Reply:
To address another comment, we are now setting per state-transition timeout. I have set it to account for the worst-case backoff grace period and expected worst-case time to attempt image pull. That comes down to 330 seconds.